### PR TITLE
Add support for `PUBSUB SHARDNUMSUB`

### PR DESCRIPTION
**Pull request by david02324**
### Description

> Describe your pull request here

very similar to the existing implementation of the PUBSUB NUMSUB command. :)

---

### Checklist

- [x] Does `npm test` pass with this change (including linting)?
- [x] Is the new or changed code fully tested?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?

### CODE REVIEW

File: packages/client/lib/commands/PUBSUB_SHARDNUMSUB.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+    if (channels) return pushVerdictArguments(args, channels);
+
+    return args;
+}
+
+export function transformReply(rawReply: Array<string | number>): Record<string, number> {
+    const transformedReply = Object.create(null);
+
+    for (let i = 0; i < rawReply.length; i +=2) {
```
Comment (younes-io) [inline packages/client/lib/commands/PUBSUB_SHARDNUMSUB.ts:19]: You can use `entries()` to iterate over the array of records.. something like:
```js
 for (const [channel, numSubscribers] of rawReply.entries()) {
    transformedReply[channel] = numSubscribers;
  }
  ```

Comment (david02324) [reply packages/client/lib/commands/PUBSUB_SHARDNUMSUB.ts:19]: Every first factor of the iterator returned by the entries() is index, so I think it is not an appropriate use 🤔

Comment (younes-io) [reply packages/client/lib/commands/PUBSUB_SHARDNUMSUB.ts:19]: How does "rawReply" look like ?

Comment (david02324) [reply packages/client/lib/commands/PUBSUB_SHARDNUMSUB.ts:19]: https://redis.io/commands/pubsub-shardnumsub/

Same as reply in pubsub numsub. Repeated channel(string) and count(number)

File: packages/client/lib/commands/PUBSUB_SHARDNUMSUB.spec.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        it('array', () => {
+            assert.deepEqual(
+                transformArguments(['1', '2']),
+                ['PUBSUB', 'SHARDNUMSUB', '1', '2']
+            );
+        });
+    });
+
+    testUtils.testWithClient('client.pubSubShardNumSub', async client => {
+        assert.deepEqual(
```
Comment (leibale) [inline packages/client/lib/commands/PUBSUB_SHARDNUMSUB.spec.ts:30]: can you please make the test "non empty" so it'll actually test `transforReply` as well?

**Review by leibale — COMMENTED** [review]
overall looks good, the tests need some changes.. :)

**Comment by leibale** [comment]
@david02324 @younes-io `redis@4.6.9`/`@redis/client@1.5.10` is on npm :tada:

**Pull request merged**